feat!: promote 2.0.0-alpha to stable (#217)
theholocron/holocron: packages/holocron-plugin-1password/README.md

  `OP_SERVICE_ACCOUNT_TOKEN` instead of the biometric flow.

Reach for one of the REST plugins (Doppler / Infisical) instead
when you want:

- Zero desktop-app dependency (Doppler-CLI on the laptop still
  needed; nothing on CI beyond a bearer token).
- 100% REST transport across every capability call (this plugin
  shells out per operation).
- A vault whose API you can reach without unlocking anything.

Both patterns are fully supported — switch by editing
`holocron.config.json`.

## Install

<!-- prettier-ignore -->
```bash
pnpm add -D @theholocron/holocron-plugin-1password@alpha

```

Requires the `op` binary on PATH (see [Prerequisite](#prerequisite)).

## Why shell-out, not REST

1Password's "REST API" is the **Connect server** — a Docker container
you have to run yourself (or pay for the cloud-hosted version). For
a solo / small-team workflow, that's overkill. The `op` CLI is what
every developer already has installed and uses for hand-debugging
anyway, so:

- **Local dev**: developer's signed-in `op` CLI (biometric unlock
  via the desktop app).
- **CI**: `OP_SERVICE_ACCOUNT_TOKEN` env var; `op` auto-detects it.
- **Either way**: same binary, same commands, same code path.

> The plugin's job is just to drive the CLI; auth handling is the

Author: ```suggestion
The plugin's job is to drive the CLI; auth handling is the
```

CLI's.

## Prerequisite

The `op` binary must be on PATH. Install:

<!-- prettier-ignore -->
```bash
brew install 1password-cli   # macOS
# or follow https://developer.1password.com/docs/cli/get-started

```

The plugin throws a clear error at construction time if `op` isn't
found.

## Auth

This plugin does NOT store a bearer token — the `op` CLI manages
its own auth via the 1Password desktop app (biometric unlock on
laptops) or `OP_SERVICE_ACCOUNT_TOKEN` on CI. `holocron auth set
1password` will accept a token, but since the plugin doesn't read
the keyring, storing one has no runtime effect. The `AUTH_HINT`
export makes this explicit in `holocron auth` output.

`holocron auth check 1password` runs `op whoami --format=json` to

> confirm you're signed in — a useful sanity check independent of

Author: ```suggestion
confirm you're signed in — a useful validation independent of
```

| `list()`              | `op item list --vault=<vault> --format=json` — names of items in the vault.        |
| `environments()`      | `op environment list --format=json` — names of 1P Environments.                    |
| `readEnvironment(id)` | `op environment read <id>` — parses `KEY=VALUE` lines into a record.               |

Plugin-level exports (not capability methods, per the auth-bootstrap
convention):

| Export        | Purpose                                                                                                     |
| ------------- | ----------------------------------------------------------------------------------------------------------- |
| `verifyToken` | `op whoami --format=json` — token arg ignored (see [Auth](#auth)). Returns `ok: true` when signed in.       |
| `AUTH_HINT`   | Explains the `op signin` / `OP_SERVICE_ACCOUNT_TOKEN` model to operators — 1P has no bearer token to store. |

### Not implemented (deliberately)

`ensureProject` / `ensureEnvironment` — 1Password's data model
doesn't have projects with sub-configs the way Doppler / Infisical
do. The vault + item hierarchy is created via the 1P UI or `op item

> create`, not via `holocron setup`. The methods are simply omitted;

Author: ```suggestion
create`, not via `holocron setup`. The methods are omitted;
```
